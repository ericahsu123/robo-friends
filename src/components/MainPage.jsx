import React, { useEffect } from 'react';

import CardList from './CardList';
import NavBar from './NavBar';
import Loading from './Loading';
import './styles.css';

const MainPage = (props) => {
  useEffect(() => {
    props.fetchRobots('https://jsonplaceholder.typicode.com/users');
    // eslint-disable-next-line
  }, []);

  const renderRobots = () => {
    if (props.robots.length === 0) {
      if (props.error) {
        return <h2>Error in fetching robots!</h2>;
      } else {
        return <Loading />;
      }
    } else {
      return <CardList />;
    }
  };
  return (
    <div className="ui container center aligned">
      <NavBar />
      <div style={{ marginTop: '120px' }}>{renderRobots()}</div>
    </div>
  );
};

export default MainPage;
