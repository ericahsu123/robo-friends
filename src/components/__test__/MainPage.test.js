import React from 'react';
// We're using our own custom render function and not RTL's render
// our custom utils also re-export everything from RTL
// so we can import fireEvent and screen here as well
import { render, cleanup } from './ test-utils';
import '@testing-library/jest-dom/extend-expect';
import MainPage from '../MainPage';

afterEach(cleanup);

describe('<MainPage/> component matches snapshots', () => {
  test('when fetching robots is still pending', () => {
    expect.assertions(1);
    const mockProps = {
      fetchRobots: jest.fn(),
      robots: [],
      error: null,
    };
    const { container } = render(<MainPage {...mockProps} />, {
      initialState: {},
    });
    expect(container).toMatchSnapshot();
  });

  test('when fetching robots is successful', () => {
    expect.assertions(1);
    const mockProps = {
      fetchRobots: jest.fn(),
      robots: [
        { id: 1, name: 'Aby', email: 'aby@gmail.com' },
        { id: 2, name: 'Abt', email: 'abt@mail.com' },
        { id: 3, name: 'Tse', email: 'tse@gmail.com' },
      ],
      error: null,
    };
    const { container } = render(<MainPage {...mockProps} />, {
      initialState: {},
    });
    expect(container).toMatchSnapshot();
  });

  test('when fetching robots fails', () => {
    expect.assertions(1);
    const mockProps = {
      fetchRobots: jest.fn(),
      robots: [],
      error: 'error in fetching robots',
    };
    const { container } = render(<MainPage {...mockProps} />, {
      initialState: {},
    });
    expect(container).toMatchSnapshot();
  });
});
