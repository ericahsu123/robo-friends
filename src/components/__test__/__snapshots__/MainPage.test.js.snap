// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainPage/> component matches snapshots when fetching robots fails 1`] = `
<div>
  <div
    class="ui container center aligned"
  >
    <div
      class="ui top fixed huge stackable menu"
    >
      <div
        class="item"
      >
        <h1>
          RoboFriends
        </h1>
      </div>
      <div
        class="right item"
      >
        <div
          class="ui icon input"
        >
          <input
            aria-label="Search Robots"
            data-testid="search-input"
            placeholder="Search robots..."
            type="text"
            value=""
          />
          <i
            class="search icon"
          />
        </div>
      </div>
    </div>
    <div
      style="margin-top: 120px;"
    >
      <h2>
        Error in fetching robots!
      </h2>
    </div>
  </div>
</div>
`;

exports[`<MainPage/> component matches snapshots when fetching robots is still pending 1`] = `
<div>
  <div
    class="ui container center aligned"
  >
    <div
      class="ui top fixed huge stackable menu"
    >
      <div
        class="item"
      >
        <h1>
          RoboFriends
        </h1>
      </div>
      <div
        class="right item"
      >
        <div
          class="ui icon input"
        >
          <input
            aria-label="Search Robots"
            data-testid="search-input"
            placeholder="Search robots..."
            type="text"
            value=""
          />
          <i
            class="search icon"
          />
        </div>
      </div>
    </div>
    <div
      style="margin-top: 120px;"
    >
      <div>
        <div
          class="ui active inverted dimmer"
        >
          <div
            class="ui text loader"
          >
            Loading
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<MainPage/> component matches snapshots when fetching robots is successful 1`] = `
<div>
  <div
    class="ui container center aligned"
  >
    <div
      class="ui top fixed huge stackable menu"
    >
      <div
        class="item"
      >
        <h1>
          RoboFriends
        </h1>
      </div>
      <div
        class="right item"
      >
        <div
          class="ui icon input"
        >
          <input
            aria-label="Search Robots"
            data-testid="search-input"
            placeholder="Search robots..."
            type="text"
            value=""
          />
          <i
            class="search icon"
          />
        </div>
      </div>
    </div>
    <div
      style="margin-top: 120px;"
    >
      <div
        class="ui cards"
      />
    </div>
  </div>
</div>
`;
